src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -1408,4 +1434,351 @@ namespace ts {
             checkNumberOfProjects(projectService, { configuredProjects: 0 });
         });
     });
+
+    describe("CompileOnSave affected list", () => {
+        function sendAffectedFileRequestAndCheckResult(session: server.Session, request: server.protocol.Request, expectedFileList: FileOrFolder[]) {
+            const actualResult = session.executeCommand(request).response;
+            const expectedFileNameList = expectedFileList.length > 0 ? ts.map(expectedFileList, f => f.path).sort() : [];
+            const actualFileNameList = actualResult.sort();
+            assert.isTrue(arrayIsEqualTo(actualFileNameList, expectedFileNameList), `Actual result is ${actualFileNameList}, while expected ${expectedFileNameList}`);
+        }
+
+        describe("for configured projects", () => {
+            let moduleFile1: FileOrFolder;
+            let file1Consumer1: FileOrFolder;
+            let file1Consumer2: FileOrFolder;
+            let moduleFile2: FileOrFolder;
+            let globalFile3: FileOrFolder;
+            let configFile: FileOrFolder;
+            let changeModuleFile1ShapeRequest1: server.protocol.Request;
+            let changeModuleFile1InternalRequest1: server.protocol.Request;
+            let changeModuleFile1ShapeRequest2: server.protocol.Request;
+            // A compile on save affected file request using file1
+            let moduleFile1FileListRequest: server.protocol.Request;
+            let host: TestServerHost;
+            let session: server.Session;
+
+            beforeEach(() => {
+                moduleFile1 = {

```

Comment:
Because sometimes they can be changed across tests. for example, oftentimes I need to change a file's content within a test, while I want the text to be the original version before starting the next test.